source/FreeRTOS_Sockets.c

```diff
@@ -4387,6 +4387,13 @@ void vSocketWakeUpUser( FreeRTOS_Socket_t * pxSocket )
         {
             pxBuffer = pxSocket->u.xTCP.txStream;
 
+            if( ( pxBuffer == NULL ) && ( pxSocket->u.xTCP.bits.bMallocError != pdTRUE ) )
+            {
```

Comment:
As `bMallocError` is an unsigned bit-field, you better use pdTRUE_UNSIGNED:
~~~diff
-    if( ( pxBuffer == NULL ) && ( pxSocket->u.xTCP.bits.bMallocError != pdTRUE ) )
+    if( ( pxBuffer == NULL ) && ( pxSocket->u.xTCP.bits.bMallocError == pdTRUE_UNSIGNED ) )
~~~
Thanks

Reply:
Thanks Hein. Updated.